masu/api/status.py

```diff
@@ -126,13 +126,11 @@ def database_status(self):
         try:
             conn = psycopg2.connect(Config.SQLALCHEMY_DATABASE_URI)
             curs = conn.cursor()
-            curs.execute('select * from pg_stat_database')
+            curs.execute('SELECT * FROM pg_stat_database')
             raw = curs.fetchall()
 
             # get pg_stat_database column names
             names = [desc[0] for desc in curs.description]
-
-            conn.close()
```

Comment:
This is what I meant 

```
finally:
    conn.close()
```

Reply:
I know what you meant. The interpreter doesn't accept it.

`conn` is defined inside the `try:` block, so it's not available in the `finally:` scope. It's local to the try scope. And I can't define outside of the `try:` block because it either A) defeats the purpose of the exception handling or B) if I do `conn = None` to initialize the variable outside of the block, the interpreter complains that `NoneType does not have a .close()`.

So, that whittles my options to leaving the `close()` call inside the `try:` portion, or deleting it entirely. ¯\_(ツ)_/¯

Reply:
That's actually not the case. The finally clause has access to anything defined in the try block. I'm guessing maybe you hit the exception and then the finally block tried to close the conn that wasn't created successfully. 

Regardless, we definitely want to close the connection. I think even cleaner than the finally block would be to use a context manager for the connection here. 

`conn = psycopg2.connect(Config.SQLALCHEMY_DATABASE_URI)` would become `with psycopg2.connect(Config.SQLALCHEMY_DATABASE_URI) as conn:`

Reply:
Context manager FTW.  That solves all of the problems.  :1st_place_medal: